spec/hooks/useProductInfo/useProductInfo.test.js

```diff
@@ -120,4 +95,63 @@ describe('Testing Hook: useProductInfo', () => {
       expect(itemSalePrice).toBeUndefined();
     });
   });
+
+  describe.each([
+    {
+      item: transformedItem,
+      itemDescription: 'a standard item',
+    },
+    {
+      item: transformedItemWithSalePrice,
+      itemDescription: 'an item on sale',
+    },
+  ])('With $itemDescription', ({ item }) => {
+    it.each([
+      ['itemId', item.itemId],
+      ['itemName', item.itemName],
+      ['itemImageUrl', item.imageUrl],
+      ['itemUrl', item.url],
+      ['itemPrice', item.data.price],
+      ['salePrice', item.data.sale_price],
+    ])('Should return the correct value for "%s"', async (property, expectedValue) => {
+      const { result } = renderHookWithCioPlp(() => useProductInfo({ item }));
+      await waitFor(() => {
+        expect(result.current[property]).toEqual(expectedValue);
+      });
+    });
+  });
+
+  describe('Testing sale price handling logic', () => {
+    it('Should return undefined salePrice if salePrice is undefined', async () => {
+      const item = transformResultItem(copyItemWithNewSalePrice(mockItemWithSalePrice, undefined));
+      const { result } = renderHookWithCioPlp(() => useProductInfo({ item }));
+      await waitFor(() => {
+        expect(result.current.salePrice).toBeUndefined();
+      });
+    });
+
+    it('Should return undefined salePrice if salePrice is negative', async () => {
+      const item = transformResultItem(copyItemWithNewSalePrice(mockItemWithSalePrice, -5));
+      const { result } = renderHookWithCioPlp(() => useProductInfo({ item }));
+      await waitFor(() => {
+        expect(result.current.salePrice).toBeUndefined();
+      });
+    });
+
+    it('Should return undefined salePrice if salePrice is greater than or equal to price', async () => {
+      const item = transformResultItem(copyItemWithNewSalePrice(mockItemWithSalePrice, Infinity));
+      const { result } = renderHookWithCioPlp(() => useProductInfo({ item }));
+      await waitFor(() => {
+        expect(result.current.salePrice).toBeUndefined();
+      });
+    });
+
+    it('Should return salePrice if salePrice is valid (positive and less than price)', async () => {
+      const item = transformResultItem(copyItemWithNewSalePrice(mockItemWithSalePrice, 1));
+      const { result } = renderHookWithCioPlp(() => useProductInfo({ item }));
+      await waitFor(() => {
+        expect(result.current.salePrice).toEqual(1);
+      });
+    });
+  });
```

Comment:
How do you feel if we reformat this to follow the `.each` pattern above? I know the test itself isn't bulky, but there's basically three lines we keep repeating for each unit test.